src/spec/payid-easy-checkout.md

```diff
@@ -0,0 +1,377 @@
+UI begins the PayID Easy Checkout flow.
+
+### PayID Easy Checkout Wallet Discovery
+The merchant UI would first assemble the PayID Discovery URL as defined in section 4.1.1 of [PAYID-DISCOVERY][],
+yielding the URL `https://wallet.com/.well-known/webfinger?resource=payid%3Aalice%24wallet.com`. 
+The merchant UI would then query the assembled URL as defined in section 4.1.2 of [PAYID-DISCOVERY][].
+
+The HTTP request in this example would look like this:
+    
+    GET /.well-known/webfinger?resource=payid%3Aalice%24wallet.com
+    Host: wallet.com
+    
+If the customer's PayID server has enabled PayID Easy Checkout in their wallet, the server would respond with something like this:
+     
+     HTTP/1.1 200 OK
+     Access-Control-Allow-Origin: *
+     Content-Type: application/jrd+json
+
+     {
+       "subject" : "payid:alice$wallet.com",
+       "links" :
+       [
+         {  
+           "rel": "https://payid.org/ns/payid-easy-checkout/1.0",
+           "template": "https://wallet.com/checkout?amount={amount}&receiverPayId={receiverPayId}&currency={currency}&nextUrl={nextUrl}"
+         }
+       ]
+     }
+
+### Expand Wallet Discovery URL Template
+The merchant UI would parse the PayID Discovery response and iterate over the "links" collection to find the link with 
+the Relation Type of "https://payid.org/ns/payid-easy-checkout/1.0". The UI can then do a search and replace on
+the "template" field value in the link, replacing all occurrences of the predefined query parameter template names with 
+the values they want to send to the customer's wallet. One query parameter of note is the "nextUrl" parameter, which
+allows the merchant to supply a redirect or callback URL for the sender's wallet to call once the customer has confirmed
+the payment. In this example, the merchant would like to display a "Thank You" page, and replaces `{nextUrl}` 
+with `https://merchant.com/thankyou`.
+
+### Redirect Customer to Their Wallet
+Once the merchant UI populates the required query parameters in the URL template, the merchant UI redirects the customer to 
+the Redirect URL so that the customer can confirm the payment.
+
+### Customer Confirms Payment
+After the customer clicks the "Pay with PayID" button the merchant's UI, and the merchant performs the previous steps,
+the customer will be redirected to the Redirect URL, which is a front end resource of the wallet. The wallet UI can
+read the query parameters from the Redirect URL and render a confirmation page or modal with all of the required fields
+pre-populated.
+
+Once the customer confirms the payment, the wallet would perform a PayID address lookup on the "receiverPayId" query
+parameter to get the payment address of the merchant and submit a transaction to the underlying ledger or payment system.
+The merchant can then redirect the user back to the URL specified in the "nextUrl" query parameter, which will display
```

Comment:
TODO: New section about the transaction Id correlation in the payID itself (e.g., a merchant can generate a custom PayID like `pay-12345$example.com` where `12345` is the `transactionId`. This unique PayID will translate into a transaction-aware layer1 address that the merchant (or its processor) can use to correlate the payment on the merchant's backend once a payment comes into that address.